internal/provider/data_source_http.go

```diff
@@ -199,6 +212,25 @@ func (d *httpDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 		Transport: clonedTr,
 	}
 
+	// Override http client with retry logic if specified
+	if !model.RetryAttempts.IsNull() && !model.RetryAttempts.IsUnknown() {
+		tflog.Debug(ctx, "HEY COUCOU TOI")
+		tflog.Debug(ctx, fmt.Sprintf("RETRY: %d %d", model.RetryAttempts.ValueInt64(), model.RetryDelay.ValueInt64()))
+		retryClient := retryablehttp.NewClient()
+		retryClient.HTTPClient.Transport = clonedTr
+		retryClient.RetryMax = int(model.RetryAttempts.ValueInt64())
+
+		delay := 0
+		if !model.RetryDelay.IsNull() && !model.RetryDelay.IsUnknown() {
+			delay = int(model.RetryDelay.ValueInt64())
+		}
+
+		retryClient.RetryWaitMin = time.Duration(delay) * time.Second
+		retryClient.RetryWaitMax = retryClient.RetryWaitMin
+		tflog.Debug(ctx, fmt.Sprintf("CLIENT: %d %d %d", retryClient.RetryMax, retryClient.RetryWaitMin, retryClient.RetryWaitMax))
```

Comment:
It might be worth adding text to make it clear what each of the values that are logged refer to.